service/utils/decorators.py

```diff
@@ -0,0 +1,19 @@
+from functools import wraps
+from firebase_admin import auth
+from firebase_admin.auth import ExpiredIdTokenError, RevokedIdTokenError, InvalidIdTokenError
+from flask import request
+
+
+def check_token(f):
+    @wraps(f)
+    def wrap(*args, **kwargs):
+        if not request.headers.get('id_token'):
+            return {'message': 'No token provided'}
+        try:
+            user = auth.verify_id_token(request.headers['id_token'])
+            user_id = request.args.get('uid')
+            assert user_id == user["user_id"]
```

Comment:
It will be more straightforward to directly pass `user["user_id"]` to controllers instead of letting the client specify it every time.

A simple way of achieving this is:

```python
kwargs.update({"uid": user["user_id"]})
return f(*args, **kwargs)
```

And each controller method that is applied by this decorator will need to have an `uid` parameter.

Reply:
Also the response should have a status code of 401 when the token is invalid

Reply:
Good idea!